subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c

```diff
@@ -231,6 +225,13 @@ static void ec_host_cmd_thread(void *hc_handle, void *arg2, void *arg3)
 	}
 }
 
+#ifndef CONFIG_EC_HOST_CMD_DEDICATED_THREAD
+void ec_host_cmd_task(void)
+{
+	ec_host_cmd_thread(&ec_host_cmd, NULL, NULL);
+}
```

Comment:
hey I think this approach is fine but I'm thinking: if we can only have ONE backend at a time, would it make sense to remove the initialization from the the backend files (the SYS_INIT and function) and instead have a single init function here, either with SYS_INIT or with a statically initialized thread? Then inside you can have a tree of `#ifdef` and only initialize the currently enabled backend. It would make things a bit easier to follow.

If you can have multiple backends... well it does not work now cause they'd try to share the stack. :-)

Reply:
Yes, you can have only one backend enabled/initialized at a time. The requirement is to have possibility to chose that backend in runtime - more than one backend are built in, but only one is initialized.

I've consider placing the `SYS_INIT`, with the #ifdefs tree here, but I decided not to do it. This is common code for every backend, I wanted to avoid adding here backend specific operations, e.g. the UART init function:
```
static int host_cmd_init(void)
{
	const struct device *const dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_host_cmd_uart_backend));

	ec_host_cmd_init(ec_host_cmd_backend_get_uart(dev));
	return 0;
}
```
there is no common code with other init functions so the entire function would have to be guarded with `#ifdef`.

IMO it would be mess and it is better to keep kind of backend specific code in separated file. You are right that `SYS_INIT` in every backend file is not ideal, but I think it is still better approach. 

What do you think? :-)

Reply:
Yeah fair enough, thanks for explaining.